LICENSE.txt

```diff
@@ -2154,89 +326,119 @@ SOFTWARE.
 
 --------------------------------------------------------------------------------
 
-The files in cpp/src/arrow/vendored/portable-snippets/ contain code from
+The files python/*/*/_version.py and python/*/*/_static_version.py
+contain code from
 
-https://github.com/nemequ/portable-snippets
+https://github.com/jbweston/miniver
 
-and have the following copyright notice:
+which is made available under the Creative Commons CC0 license.
 
-Each source file contains a preamble explaining the license situation
-for that file, which takes priority over this file.  With the
-exception of some code pulled in from other repositories (such as
-µnit, an MIT-licensed project which is used for testing), the code is
-public domain, released using the CC0 1.0 Universal dedication (*).
+--------------------------------------------------------------------------------
 
-(*) https://creativecommons.org/publicdomain/zero/1.0/legalcode
+3rdparty dependency libpq is statically linked in certain binary
+distributions, like the python wheels. libpq has the following license:
 
---------------------------------------------------------------------------------
+Portions Copyright © 1996-2022, The PostgreSQL Global Development Group
 
-The files in cpp/src/arrow/vendored/fast_float/ contain code from
+Portions Copyright © 1994, The Regents of the University of California
 
-https://github.com/lemire/fast_float
+Permission to use, copy, modify, and distribute this software and its
+documentation for any purpose, without fee, and without a written
+agreement is hereby granted, provided that the above copyright notice
+and this paragraph and the following two paragraphs appear in all
+copies.
+
+IN NO EVENT SHALL THE UNIVERSITY OF CALIFORNIA BE LIABLE TO ANY PARTY
+FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND
+ITS DOCUMENTATION, EVEN IF THE UNIVERSITY OF CALIFORNIA HAS BEEN
+ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-which is made available under the Apache License 2.0.
+THE UNIVERSITY OF CALIFORNIA SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE
+PROVIDED HEREUNDER IS ON AN "AS IS" BASIS, AND THE UNIVERSITY OF
+CALIFORNIA HAS NO OBLIGATIONS TO PROVIDE MAINTENANCE, SUPPORT,
+UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
 
 --------------------------------------------------------------------------------
 
-The file python/pyarrow/vendored/docscrape.py contains code from
-
-https://github.com/numpy/numpydoc/
+3rdparty dependency lz4 is statically linked in certain binary
```

Comment:
For lz4, vcpkg, zlib, I added them because they do get bundled into our wheels: https://github.com/apache/arrow-adbc/actions/runs/3653243387/jobs/6172490942#step:6:478

```
The following packages will be built and installed:
    libpq[core,lz4,openssl,zlib]:x64-linux-static-release -> 14.4#2
  * lz4[core]:x64-linux-static-release -> 1.9.3#4
  * openssl[core]:x64-linux-static-release -> 3.0.7
  * vcpkg-cmake[core]:x64-linux -> 2022-10-30
  * vcpkg-cmake-config[core]:x64-linux -> 2022-02-06#1
  * vcpkg-cmake-get-vars[core]:x64-linux -> 2022-09-04
  * zlib[core]:x64-linux-static-release -> 1.2.13
```

Reply:
(to be fair, those vcpkg ones might be build-only, I'll take a closer look in a moment - and I'll figure out how to make CI build the wheels with the linker flags exposed)

Reply:
Removed.